rest/route/stats.go

```diff
@@ -484,3 +487,51 @@ func (tsh *taskStatsHandler) Run(ctx context.Context) gimlet.Responder {
 func makeGetProjectTaskStats(sc data.Connector) gimlet.RouteHandler {
 	return &taskStatsHandler{sc: sc}
 }
+
+type cedarTestStatsMiddleware struct{}
+
+func CheckCedarTestStats() gimlet.Middleware {
+	return &cedarTestStatsMiddleware{}
+}
+
+func (m *cedarTestStatsMiddleware) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	ctx := r.Context()
+
+	newURL := fmt.Sprintf(
+		"https://cedar.mongodb.com/rest/v1/historical_test_data/%s?%s",
+		gimlet.GetVars(r)["project_id"],
+		r.URL.RawQuery,
+	)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, newURL, nil)
+	if err != nil {
+		gimlet.WriteResponse(rw, gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "problem creating cedar test stats request")))
+		return
+	}
+
+	c := utility.GetHTTPClient()
+	defer utility.PutHTTPClient(c)
+
+	resp, err := c.Do(req)
+	if err != nil {
+		gimlet.WriteResponse(rw, gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "problem sending test stats request to cedar")))
```

Comment:
I'm not a REST middleware expert, but I don't think it's valid to write a response and not return, because writing multiple responses (i.e. the result of this and the result of the next middleware) to a request probably won't be handled properly.

Reply:
Added a return after